master/buildbot/process/buildstep.py

```diff
@@ -844,7 +844,11 @@ def makeRemoteShellCommand(self, collectStdout=False, collectStderr=False,
         kwargs.update(overrides)
         stdio = None
         if stdioLogName is not None:
-            stdio = yield self.addLog(stdioLogName)
+            try:
```

Comment:
A comment that describes why the same operation is performed twice would be nice.